study.md

```diff
@@ -44,21 +49,30 @@ what do developers mean when they say "view-state"? How do front-end "routers"
 help manage view-state?
 
 ```md
-<!-- your answer here -->
+The user's understanding of a "page" translates directly to the developer's
+concept of a view. With single page apps, the user's "pages" are just
+different "views" supported in the single HTML.
+
+I think view-state is a snapshot of what the view looks like with all the
+applicable data. Users expect their applications to automatically work with
+their browser's back button and forward button (front-end "routers").
 ```
 
 ## Define the advantages of an SPA over a traditional application
 
 What are the advantages of a single page application over a traditional appliction?
 
 ```md
-<!-- your answer here -->
+SPAs are more responsive. In older apps where each view was an HTML file
+on a server, rendering those pages required a roundtrip call to the server.
+Single page apps don't need to do that since the initial page is all it needs.
+
 ```
 
 ## List some technologies used to make SPA's
 
 What are some of the tools used to create single page applications?
 
 ```md
-<!-- your answer here -->
+html5, css, javascript, ajax
```

Comment:
Yes but most important are front end frameworks like Angular, Ember, and React that are great tools for managing view states and front end routing. Although you managed to do it pretty well on your tic tac toe project without any help of a framework. Super impressive!